x/http_test.go

```diff
@@ -28,27 +28,48 @@ func TestRequestURL(t *testing.T) {
 	}).String(), "http://foobar/foo")
 }
 
-func TestAcceptToRedirectOrJSON(t *testing.T) {
-	wr := herodot.NewJSONWriter(logrusx.New("", ""))
+func TestAcceptsContentType(t *testing.T) {
+	assert.True(t, AcceptsContentType(&http.Request{
+		Header: map[string][]string{"Accept": {"application/json"}},
+	}, "application/json"))
+	assert.True(t, AcceptsContentType(&http.Request{
+		Header: map[string][]string{"Accept": {"application/json; charset=utf-8"}},
+	}, "application/json"))
+	assert.False(t, AcceptsContentType(&http.Request{
+		Header: map[string][]string{"Accept": {"application/json"}},
+	}, "text/html"))
+	assert.False(t, AcceptsContentType(&http.Request{
+		Header: map[string][]string{"Accept": {"application/json; charset=utf-8"}},
+	}, "text/html"))
+	assert.True(t, AcceptsContentType(&http.Request{
+		Header: map[string][]string{"Accept": {"text/html"}},
+	}, "text/html"))
+	assert.True(t, AcceptsContentType(&http.Request{
+		Header: map[string][]string{"Accept": {"text/html, application/json;q=0.9, */*;q=0.8"}},
+	}, "application/json"))
```

Comment:
Please add a test for:

```
assert.True(t, AcceptsContentType(&http.Request{
  Header: map[string][]string{"Accept": {"application/json;q=0.9, */*;q=0.8"}},
}, "text/html"))
assert.True(t, AcceptsContentType(&http.Request{
  Header: map[string][]string{"Accept": {"application/json;q=0.9, text/html;q=0.8, */*;q=0.8"}},
}, "text/html"))
```

And also different spacing and numeric variations:

```
 application/json;q=0.9
 application/json; q=0.9
 application/json; q=0.9,*/*;q=1
 application/json; q=0.9, */*;q=1
 application/json; q=0.9 ,*/*;q=1
```